src/System.Private.CoreLib/src/System/Reflection/Assembly.CoreCLR.cs

```diff
@@ -51,7 +51,9 @@ private static Assembly LoadFromResolveHandler(object sender, ResolveEventArgs a
             // in the same folder as it.
             // Form the name of the assembly using the path of the assembly that requested its load.
             AssemblyName requestedAssemblyName = new AssemblyName(args.Name);
-            string requestedAssemblyPath = Path.Combine(Path.GetDirectoryName(requestorPath), requestedAssemblyName.Name+".dll");
+            string requestedAssemblyPath = String.IsNullOrEmpty(requestedAssemblyName.CultureName) ?
+                Path.Combine(Path.GetDirectoryName(requestorPath), requestedAssemblyName.Name+".dll") :
```

Comment:
That fallback logic is in the outer loop.  This should be called with "en-US", then "en", then the fallback neutral culture if and only if that culture is configured to be in a satellite assembly.

I think this will just work.  I will take the time to test it in detail tomorrow.